yt/utilities/io_handler.py

```diff
@@ -172,20 +174,52 @@ def _read_particle_data_file(self, data_file, ptf, selector=None):
     def _read_particle_selection(
         self, chunks, selector, fields: List[Tuple[str, str]]
     ) -> Dict[Tuple[str, str], np.ndarray]:
-        data: Dict[Tuple[str, str], List[np.ndarray]] = {}
 
-        # Initialize containers for tracking particle, field information
+        # re-organize the fields to read
+        ptf, field_maps = self._get_particle_field_containers(fields)
+        data: dict = {field: [] for field in fields}
```

Comment:
Actually having a type hint here is a very good idea, would you mind fleshing it out a little ?